stdlib/public/runtime/BuiltinProtocolConformances.cpp

```diff
@@ -30,7 +30,39 @@ using StaticInfixWitness = SWIFT_CC(swift) bool(OpaqueValue *, OpaqueValue *,
 
 // Elf indirect symbol references.
 #if defined(__ELF__)
-#define INDIRECT_RELREF_GOTPCREL(SYMBOL) SYMBOL "@GOTPCREL + 1"
+#define INDIRECT_RELREF_GOTPCREL(SYMBOL) ".Lgot." SYMBOL " - . + 1"
+
+// Helper for defining the .type on linux.
+#if defined(__arm__) && !defined(__aarch64__)
+#define TYPE_OBJECT "%object"
+#else
+#define TYPE_OBJECT "@object"
+#endif
+
+// Helper to get the pointer sized pointer to symbol and the right alignment.
+#if defined(__x86_64__) || defined(__aarch64__)
+#define POINTER_FOR_GOT(SYMBOL) \
+  "  .quad (" SYMBOL ")\n" \
+  "  .size .Lgot." SYMBOL ", 8\n"
```

Comment:
I think you might want to have a level of indirection here.  The `.L` not guaranteed to the assembler local prefix.